src/iplant_groups/service/groups.clj

```diff
@@ -35,12 +36,14 @@
 
 (defn update-group-privileges
   [group-name {:keys [updates]} {:keys [user replace] :or {replace true} :as params}]
+  (when replace (util/verify-not-removing-own-privileges user (map :subject_id updates)))
   (doseq [[privileges vs] (group-by (comp set :privileges) updates)]
     (grouper/update-group-privileges user replace group-name (mapv :subject_id vs) privileges))
   (get-group-privileges group-name params))
 
 (defn remove-group-privileges
   [group-name {:keys [updates]} {:keys [user] :as params}]
+  (util/verify-not-removing-own-privileges user (map :subject_id updates))
```

Comment:
We also have the endpoint to leave a team.  Will that still work for admins if they, for whatever reason, decide to leave their own team?